app/services/messages/service.go

```diff
@@ -308,7 +308,8 @@ func (ss *Service) verifySignature(authMsgBytes []byte, signatureBytes []byte, t
 
 // awaitTransfer checks until given transfer is found
 func (ss *Service) awaitTransfer(transferID string) (*entity.Transfer, error) {
-	for {
+	i := 0
+    for i < 30 {
```

Comment:
nit: can this 30 be in a constant/var above describing that it's the retry attempt count